### PR TITLE
Add support composer 2 and php 8.0

### LINKED ISSUE #30: Update composer to v2 and add support for PHP 8
### Is your feature request related to a problem?

> Composer 2 and php 8 availeble now

### Describe the solution you'd like

> Add support for PHP 8 and update composer to v2

### PR DESCRIPTION
**Pull request by 4irik**
## Description

Add support composer 2 and php 8.0

Fixes #30

## Checklist

- [ ] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I wrote unit tests for my code _(if tests is required for my changes)_
- [x] I have made changes in `CHANGELOG.md` file

### CODE REVIEW

**Comment by blackbass1988** [comment]
поставь все галочки)

File: Dockerfile [hunk 1]
```diff
@@ -4,7 +4,7 @@ ENV \
     COMPOSER_ALLOW_SUPERUSER="1" \
```
Comment (IgorKorytin) [inline Dockerfile:4]: This env is unused in composer 2

File: Dockerfile [hunk 2]
```diff
@@ -13,7 +13,6 @@ RUN set -x \
     && pecl install xdebug-2.9.6 1>/dev/null \
     && apk del .build-deps \
     && mkdir /src ${COMPOSER_HOME} \
-    && composer global require 'hirak/prestissimo' --no-interaction --no-suggest --prefer-dist \
```
Comment (IgorKorytin) [inline Dockerfile:16]: Remove it from github actions workflow too please

File: composer.json [hunk 3]
```diff
@@ -15,16 +15,16 @@
         }
     ],
     "require": {
-        "php": "^7.2",
+        "php": "^7.2 || ^8.0",
```
Comment (IgorKorytin) [inline composer.json:18]: change php version in `Dockerfile` and add php 8 to github actions

File: Dockerfile [hunk 4]
```diff
@@ -13,7 +13,6 @@ RUN set -x \
     && pecl install xdebug-2.9.6 1>/dev/null \
```
Comment (IgorKorytin) [inline Dockerfile:13]: maybe update it to v 3?

**Review by IgorKorytin — CHANGES_REQUESTED** [review]
1. Remove `--no-suggest` key from composer scripts where key `-n` used (makefile, github actions)
2. Add `-e 'XDEBUG_MODE=coverage'` to `make test-cover` script, and add this env to github actions
3.  Do you check `make test-cover` scpipt on PHP 8 and `latest` config? Think it will not work or u'll get a deprecated warning

**Comment by eldario** [comment]
@4irik 
Please, make some changes by threads

**Comment by eldario** [comment]
No activity

**Pull request closed without merging**